src/Payum/Core/Bridge/Guzzle/HttpClientFactory.php

```diff
@@ -3,31 +3,20 @@
 
 use GuzzleHttp\Client as GuzzleClient;
 use GuzzleHttp\ClientInterface as GuzzleClientInterface;
+use Http\Adapter\Guzzle6\Client;
+use Http\Discovery\HttpClientDiscovery;
 use Payum\Core\HttpClientInterface;
 
+/**
+ * @deprecated This will be removed in 2.0. Consider using Http\Discovery\HttpClientDiscovery.
+ */
 class HttpClientFactory
 {
     /**
-     * @return GuzzleClientInterface
-     */
-    public static function createGuzzle()
-    {
-        $curl = curl_version();
-
-        $curlOptions = [
-            CURLOPT_USERAGENT => sprintf('Payum/1.x curl/%s PHP/%s', $curl['version'], phpversion()),
-        ];
-
-        return new GuzzleClient([
-            'curl' => $curlOptions
-        ]);
-    }
-
-    /**
      * @return HttpClientInterface
      */
     public static function create()
     {
-        return new HttpClient(static::createGuzzle());
+        return new HttpClient(HttpClientDiscovery::find());
```

Comment:
So we can drop a guzzle as default client, since users are forced to take care of this on composer update.